File: src/main/java/no/nav/syfo/rest/api/model/RSReservasjon.java
package no.nav.syfo.rest.api.model;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

@Data
@Accessors(fluent = true)
@EqualsAndHashCode
public class RSReservasjon {


    public enum KontaktInfoFeilAarsak {
        RESERVERT,
        KODE6,
        INGEN_KONTAKTINFORMASJON,
        UTGAATT
    }

    public Boolean skalHaVarsel;
    public KontaktInfoFeilAarsak feilAarsak;
}

File: src/main/kotlin/no/nav/syfo/controller/internad/person/RSReservasjon.kt
package no.nav.syfo.controller.internad.person

data class RSReservasjon (
    val skalHaVarsel: Boolean? = null,
    val feilAarsak: KontaktInfoFeilAarsak? = null
)

enum class KontaktInfoFeilAarsak {
    RESERVERT,
    KODE6,
    INGEN_KONTAKTINFORMASJON,
    UTGAATT
}
